packages/protocol-kit/src/Safe.ts

```diff
@@ -1486,7 +1534,11 @@ class Safe {
     const transactions = [safeDeploymentBatchTransaction, safeBatchTransaction]
 
     // this is the transaction with the batch
-    const safeDeploymentBatch = await this.createTransactionBatch(transactions, transactionOptions)
+    const safeDeploymentBatch = await this.createTransactionBatch(
+      transactions,
+      transactionOptions,
+      true // include the on chain identifier
```

Comment:
I don't understand this. Why including it explicetely? If the private variable exists then we can include it right?

Reply:
It’s because we don’t always know where the batch created with the `createTransactionBatch` function will be used. Automatically appending the trackId could cause issues in transactions that aren’t meant to be executed directly.